src/SwapMath.sol

```diff
@@ -37,133 +66,117 @@ library SwapMath {
             uint256 feeAmount
         )
     {
-        bool zeroForOne = sqrtRatioCurrentX96 >= sqrtRatioTargetX96;
-        bool exactIn;
-        uint256 amountRemainingAbs;
-        assembly {
```

Comment:
The one on the left is more readable and has less code duplicates. How much gas are we saving with the version on the right?

Reply:
How are more booleans and branching more readable? Hundreds of gas is saved by simply refactoring.
https://github.com/Uniswap/v4-core/pull/280/files#diff-4f433dfe3cc381d5daee1e814f73c5acf705fe4218a113fb79360a1ef331bfa2

Reply:
The one on the left is more readable because the logic are separated into different parts. What we are doing on the right is to pack all those logics together tightly. That makes readability really poor.

Reply:
The logic is clearly separated into the `exactIn` and `exactOut` cases without duplicate boolean checks. Observe the clear logic in one piece without using the diff view.

Reply:
SG